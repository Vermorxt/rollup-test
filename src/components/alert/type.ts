import { Ui_colors } from '../_types/colors/ui_colors'
import { HTMLProps, MouseEventHandler } from 'react'

export interface Ui_AlertProps extends HTMLProps<HTMLDivElement>, Ui_colors {
  noIcon?: boolean
  onDenyButtonText?: string
  onAcceptButtonText?: string
  onDeny?: MouseEventHandler<HTMLButtonElement> | undefined
  onAccept?: MouseEventHandler<HTMLButtonElement> | undefined
}
