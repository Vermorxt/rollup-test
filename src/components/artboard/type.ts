import { Ui_colors } from './../_types/colors/ui_colors'
import { HTMLProps } from 'react'

export interface Ui_ArtboardProps extends HTMLProps<HTMLSpanElement>, Ui_colors {
  phone1?: boolean
  phone2?: boolean
  phone3?: boolean
  phone4?: boolean
  phone5?: boolean
  phone6?: boolean
  horizontal?: boolean
  demo?: boolean
}
