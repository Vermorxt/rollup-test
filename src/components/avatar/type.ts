import { HTMLProps } from 'react'
import { Ui_Sizes } from '../_types/sizes/sizes'

interface Ui_Avatar_radius {
  roundedXl?: boolean
  rounded2xl?: boolean
  rounded3xl?: boolean
  roundedLg?: boolean
  roundedFull?: boolean
}

interface Ui_Avatar_ring {
  ring?: boolean
  ringPrimary?: boolean
  ringSecondary?: boolean
  ringOffset?: '0' | '1' | '2' | '4' | '8'
  ringOffsetBackground?: string // NOTE: https://tailwindcss.com/docs/ring-offset-color
}

interface Ui_Avatar_shapes {
  squircle?: boolean
  hexagon?: boolean
  triangle?: boolean
}

interface Ui_Avatar_online {
  online?: boolean
  offline?: boolean
}

export interface Ui_AvatarProps
  extends HTMLProps<HTMLDivElement>,
    Ui_Sizes,
    Ui_Avatar_radius,
    Ui_Avatar_shapes,
    Ui_Avatar_online,
    Ui_Avatar_ring {
  src?: string
  usePlaceHolder?: string // NOTE: content of last avatar item (e.g.: +99)
}
