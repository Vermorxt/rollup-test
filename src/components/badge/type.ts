import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
import { Ui_Sizes } from '../_types/sizes/sizes'

export type TooltipPositionProps = 'top' | 'left' | 'bottom' | 'right'

export interface Ui_Badge_variants {
  outline?: boolean
  active?: boolean
  disabled?: boolean
  glass?: boolean
}

export interface Ui_Badge_singleClass {
  loading?: boolean
  noAnimation?: boolean
}

export interface Ui_BadgeProps
  extends HTMLProps<HTMLSpanElement>,
    Ui_colors,
    Ui_Sizes,
    Ui_Badge_variants,
    Ui_Badge_singleClass {
  wide?: boolean
  full?: boolean
}
