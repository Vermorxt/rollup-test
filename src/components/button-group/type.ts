import { HTMLProps } from 'react'

export interface Ui_ButtonGroupProps extends HTMLProps<HTMLDivElement> {}
