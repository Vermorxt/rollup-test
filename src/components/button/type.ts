import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
import { Ui_Sizes } from '../_types/sizes/sizes'
import { TooltipProps } from '../_types/tooltip/tooltip-props'

export interface Ui_Button_variants {
  outline?: boolean
  active?: boolean
  disabled?: boolean
  glass?: boolean
}

export interface Ui_Button_singleClass {
  loading?: boolean
  noAnimation?: boolean
}

export interface Ui_Button_sizes extends Ui_Sizes {
  wide?: boolean
  block?: boolean
  circle?: boolean
  square?: boolean
  full?: boolean
}

export type Ui_ButtonType = 'button' | 'submit' | 'reset' | undefined

export interface Ui_ButtonProps
  extends HTMLProps<HTMLButtonElement>,
    Ui_colors,
    Ui_Button_sizes,
    Ui_Button_variants,
    Ui_Button_singleClass {
  tooltip?: TooltipProps
  type?: Ui_ButtonType
}
