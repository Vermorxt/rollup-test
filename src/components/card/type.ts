import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
interface Ui_Card_style {
  bgPrimary?: boolean
  bgSecondary?: boolean
  bgNeutral?: boolean
  textPrimaryContent?: boolean
  textSecondaryContent?: boolean
  textNeutralContent?: boolean
  bgBase?: '100' | '200' | '300'
}
export interface Ui_CardProps extends HTMLProps<HTMLDivElement>, Ui_Card_style {
  imageAsBackground?: boolean
  compact?: boolean
  itemsCenter?: boolean
  textCenter?: boolean
  glass?: boolean
  cardSide?: boolean
  cardSideLarge?: boolean
  w96?: boolean
  actionCenter?: boolean
  actionStart?: boolean
  actionSpaceBetween?: boolean
}
