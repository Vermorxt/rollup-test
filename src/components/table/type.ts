import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
export interface Ui_TableProps extends HTMLProps<HTMLTableElement>, Ui_colors {
  active?: boolean
  hover?: boolean
  zebra?: boolean
  stickyHeader?: boolean
  compact?: boolean
  bgColor?: 'primary' | 'secondary'
  textColor?: 'primary' | 'secondary'
}
