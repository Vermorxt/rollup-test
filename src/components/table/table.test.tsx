import '@testing-library/jest-dom'
import { FC } from 'react'
import { SpecialProps } from '../../_tests-utils/it-should-have-class'
import { testGenerator, TestGeneratorProps } from '../../_tests-utils/_utils/test-generator'
import { Ui_TableProps } from './type'
import Ui_Table from './table'
import React from 'react'

const componentName = Ui_Table
const selector = 'table'
const isTable = 'table'
const refType = HTMLTableElement
const defaultProps: Ui_TableProps = {}

interface SpecialPropsMerged extends SpecialProps {
  className: string
  prop?: Ui_TableProps
}

const component = (<Ui_Table {...defaultProps} ref={null}></Ui_Table>) as unknown as FC<{}>

const testName = `@cms/core/${'table'}`

const componentPropsColors: SpecialPropsMerged[] = [
  { prop: { primary: true }, className: `bg-primary.text-primary-content` },
  { prop: { secondary: true }, className: `bg-secondary.text-secondary-content` },
  { prop: { accent: true }, className: `bg-accent.text-accent-content` },
  { prop: { info: true }, className: `bg-info.text-info-content` },
  { prop: { success: true }, className: `bg-success.text-success-content` },
  { prop: { warning: true }, className: `bg-warning.text-warning-content` },
  { prop: { error: true }, className: `bg-error.text-error-content` },
  { prop: { neutral: true }, className: `bg-neutral.text-neutral-content` },
  { prop: { ghost: true }, className: `bg-ghost.text-ghost-content` },
  { prop: { link: true }, className: `bg-link.text-link-content` },
]

const componentPropsDefault: SpecialPropsMerged[] = [
  { prop: { bgColor: 'primary' }, className: `bg-${'primary'}.text-${'primary'}-content` },
  { prop: { bgColor: 'secondary' }, className: `bg-${'secondary'}.text-${'secondary'}-content` },
  { prop: { compact: true }, className: `${selector}-compact` },
]

const componentProps = [...componentPropsColors, ...componentPropsDefault]

const testConfig = {
  testName,
  componentName,
  component,
  componentProps,
  defaultProps,
  selector,
  isTable,
  refType,
} as TestGeneratorProps

testGenerator(testConfig)
