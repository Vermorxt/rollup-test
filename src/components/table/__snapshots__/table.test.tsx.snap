// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@cms/core/table should create snapshot 1`] = `
<div>
  <table
    class="table w-full 
        
        
        
        
        
        
        
        
        
        
        
        
        
        
      "
    tabindex="0"
  />
</div>
`;
